### PR TITLE
Create a separate go module for pinniped-auth plugin

### LINKED ISSUE #3421: Create a separate go module for pinniped-auth plugin
**Describe the feature request**
Currently pinniped-auth plugin is using common tanzu-framework go.mod file for build. We would like to decouple that and create plugin specific go.mod file within plugin directory.

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- (X) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by prkalle**
Signed-off-by: Prem Kumar Kalle <pkalle@vmware.com>

### What this PR does / why we need it
This PR creates a separate go module for pinniped-auth plugin

### Which issue(s) this PR fixes

Fixes #3421 

### Release note

```release-note
None
```

### CODE REVIEW

File: cmd/cli/plugin/pinniped-auth/go.mod [hunk 1]
```diff
@@ -0,0 +1,69 @@
+module github.com/vmware-tanzu/tanzu-framework/plugin/pinniped-auth
+
+go 1.17
+
+replace (
+	github.com/vmware-tanzu/tanzu-framework/apis/config => ../../../../apis/config
```
Comment (anujc25) [inline cmd/cli/plugin/pinniped-auth/go.mod:6]: I don't see usage of this module. Can we remove this if that is the case?

Comment (prkalle) [reply cmd/cli/plugin/pinniped-auth/go.mod]: Fixed. Thanks

**Review by marckhouzam — APPROVED** [review]
The tests fail but the code LGTM...

**Review by anujc25 — APPROVED** [review]
LGTM.

**Pull request merged**